### PR TITLE
Change models and repositories

### PR DESCRIPTION
**Pull request by C0Rs1K**

### CODE REVIEW

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 1]
```diff
@@ -62,6 +43,14 @@ public CurtainsModel()
...
         /// </summary>
         public string Purpose { get; set; }
+        /// <summary>
+        ///  This property represents ColorId column.
+        /// </summary>
+        public int ColorId { get; set; }
+        /// <summary>
+        ///  This property represents BrasingId column.
+        /// </summary>
+        public int BrasingId { get; set; }
```
Comment (Pilyut) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: Maybe bracingId instead of brasingId :)

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 2]
```diff
@@ -86,17 +75,29 @@ public CurtainsModel()
         /// </summary>
         public virtual CurtainsKindModel CurtainsKind { get; set; }
         /// <summary>
+        ///  This property represents of Bracing.
+        /// </summary>
+        public virtual BracingModel Brasing { get; set; }
```
Comment (Pilyut) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: The same typo :)

**Review by Pilyut — APPROVED** [review]

File: Curtains/.vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (EgorAntonovich) [inline Curtains/.vscode/launch.json]: Why you add .vscode?

File: Curtains/Curtains.Domain/Models/AccessoriesModel.cs [hunk 4]
```diff
@@ -28,13 +22,9 @@ public AccessoriesModel()
...
         public virtual ColorModel Color { get; set; }
         /// <summary>
-        ///  This property represents list of UserOrders in which there are accessoreis.
-        /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
-        /// <summary>
         ///  This property represents list of curtains in which there are accessories.
         /// </summary>
-        public virtual ICollection<CurtainsModel> Curtains { get; set; }
+        public virtual ICollection<CurtainModel> Curtains { get; set; } = new List<CurtainModel>();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/AccessoriesModel.cs]: Why you define new List?

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 5]
```diff
@@ -31,40 +19,24 @@ public BedspreadsModel()
...
-        /// <summary>
         ///  This property represents fabric of bedspreads.
         /// </summary>
         public virtual FabricModel Fabric { get; set; }
         /// <summary>
-        ///  This property represents list of product sets, in which there are bedspreads.
+        ///  This property represents ProductImage of bedspreads.
         /// </summary>
-        public virtual ICollection<ProductSetModel> Sets { get; set; }
+        public virtual ICollection<ProductImageModel> ProductImages { get; } = new List<ProductImageModel>();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: same

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 6]
```diff
@@ -31,40 +19,24 @@ public BedspreadsModel()
...
         /// <summary>
-        ///  This property represents list of UserOrders, in which there are bedspreads.
+        ///  This property represents product sets with bedspreads.
         /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
+        public virtual ICollection<ProductSetModel> ProductSets { get; } = new List<ProductSetModel>();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: same

File: Curtains/.vscode/tasks.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (EgorAntonovich) [inline Curtains/.vscode/tasks.json]: Ignore .vscode

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 8]
```diff
@@ -1,70 +1,42 @@
...
-            Sets = new HashSet<ProductSetModel>();
-            UserOrders = new HashSet<UserOrderModel>();
-        }
-
         #region PropertiesRegion
         /// <summary>
         ///  This property represents size of bedspreads.
         /// </summary>
-        public string Size { get; set; }
+        public string Size { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: remove null
it should be only `public string Size { get; set; }`

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 9]
```diff
@@ -1,70 +1,42 @@
...
         /// <summary>
         ///  This property represents purpose of bedspreads.
         /// </summary>
-        public string Purpose { get; set; }
+        public string Purpose { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: same

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 10]
```diff
@@ -1,70 +1,42 @@
...
         #region NavigationProperties
         /// <summary>
-        ///  This property represents marketing information of bedspreads.
-        /// </summary>
-        public MarketingInfoModel MarketingInfo { get; set; }
-        /// <summary>
         ///  This property represents fabric of bedspreads.
         /// </summary>
-        public virtual FabricModel Fabric { get; set; }
+        public virtual FabricModel Fabric { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: the same

File: Curtains/Curtains.Domain/Models/ColorModel.cs [hunk 11]
```diff
@@ -1,43 +1,31 @@
...
-            Accessories = new HashSet<AccessoriesModel>();
-            Lambrequins = new HashSet<LambrequinsModel>();
-        }
-
         #region PropertiesRegion
         /// <summary>
         ///  This property represents code of color.
         /// </summary>
-        public string Code { get; set; }
+        public string Code { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/ColorModel.cs]: the same

File: Curtains/Curtains.Domain/Models/ColorModel.cs [hunk 12]
```diff
@@ -1,43 +1,31 @@
...
+        /// <summary>
+        ///  This property represents code of color group.
+        /// </summary>
+        public string ColorGroup { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/ColorModel.cs]: the same

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 13]
```diff
@@ -61,42 +42,58 @@ public CurtainsModel()
         /// <summary>
         ///  This property represents purpose of curtains.
         /// </summary>
-        public string Purpose { get; set; }
+        public string Purpose { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: the same

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 14]
```diff
@@ -61,42 +42,58 @@ public CurtainsModel()
...
+        /// </summary>
+        public int BraсingId { get; set; }
         #endregion
 
         #region NavigationProperties
         /// <summary>
         ///  This property represents fabric of curtains.
         /// </summary>
-        public virtual FabricModel Fabric { get; set; }
+        public virtual FabricModel Fabric { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: the same

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 15]
```diff
@@ -61,42 +42,58 @@ public CurtainsModel()
...
         /// <summary>
         ///  This property represents material of curtains.
         /// </summary>
-        public virtual MaterialModel Material { get; set; }
+        public virtual MaterialModel Material { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: the same

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 16]
```diff
@@ -61,42 +42,58 @@ public CurtainsModel()
...
         /// <summary>
         ///  This property represents accessories of curtains.
         /// </summary>
-        public virtual AccessoriesModel Accessories { get; set; }
+        public virtual AccessoriesModel Accessories { get; set; } = null!;
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: the same

File: Curtains/Curtains.Infrastructure/Repositories/BracingRepository.cs [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        #region MethodsRegion
+        /// <summary>
+        /// This method get all<c> BracingModel <c> entities from database with includes tabels by entity Id
+        /// </summary>
+        /// <returns>Collection of BracingModel entities in List type</return>
+        public IEnumerable<BracingModel> GetAll()
+        {
+            return _curtainsContext.Bracings.AsNoTracking().AsEnumerable();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Infrastructure/Repositories/BracingRepository.cs:41]: add empty or null check anyway

File: Curtains/Curtains.Infrastructure/Repositories/DesignRepository.cs [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        #region MethodsRegion
+        /// <summary>
+        /// This method get all<c> DesignModel <c> entities from database with includes tabels by entity Id
+        /// </summary>
+        /// <returns>Collection of DesignModel entities in List designs</return>
+        public IEnumerable<DesignModel> GetAll()
+        {
+            return _curtainsContext.Designs.AsNoTracking().AsEnumerable();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Infrastructure/Repositories/DesignRepository.cs:41]: empty or null check. up to you

File: Curtains/Curtains.Infrastructure/Repositories/MetaDataRepository.cs [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        #region MethodsRegion
+        /// <summary>
+        /// This method get all<c> MetadataModel <c> entities from database with includes tabels by entity Id
+        /// </summary>
+        /// <returns>Collection of MetadataModel entities in List type</return>
+        public IEnumerable<MetaDataModel> GetAll()
+        {
+            return _curtainsContext.MetaDatas.AsNoTracking().AsEnumerable();
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Infrastructure/Repositories/MetaDataRepository.cs:41]: empty or null check

**Review by EgorAntonovich — CHANGES_REQUESTED** [review]

**Review by EgorAntonovich — APPROVED** [review]

**Comment by EgorAntonovich** [comment]
@zd-prog please review it.

**Comment by EgorAntonovich** [comment]
@C0Rs1K please fix conflicts

File: Curtains/Curtains.Domain/Models/AccessoriesModel.cs [hunk 20]
```diff
@@ -5,17 +5,11 @@
...
-            Curtains = new HashSet<CurtainsModel>();
-            UserOrders = new HashSet<UserOrderModel>();
-        }
-
         #region PropertiesRegion
         /// <summary>
         ///  This property represents the ColorId column.
         /// </summary>
-        public int ColorId { get; set; }
+        public int? ColorId { get; set; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/AccessoriesModel.cs]: Why did you make ColorID Nullable?

File: Curtains/Curtains.Domain/Models/AccessoriesModel.cs [hunk 21]
```diff
@@ -28,13 +22,9 @@ public AccessoriesModel()
...
         public virtual ColorModel Color { get; set; }
         /// <summary>
-        ///  This property represents list of UserOrders in which there are accessoreis.
-        /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
-        /// <summary>
         ///  This property represents list of curtains in which there are accessories.
         /// </summary>
-        public virtual ICollection<CurtainsModel> Curtains { get; set; }
+        public virtual ICollection<CurtainModel>? Curtains { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/AccessoriesModel.cs]: This Collection can be empty, not null

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 22]
```diff
@@ -31,40 +19,24 @@ public BedspreadsModel()
...
-        /// <summary>
         ///  This property represents fabric of bedspreads.
         /// </summary>
         public virtual FabricModel Fabric { get; set; }
         /// <summary>
-        ///  This property represents list of product sets, in which there are bedspreads.
+        ///  This property represents ProductImage of bedspreads.
         /// </summary>
-        public virtual ICollection<ProductSetModel> Sets { get; set; }
+        public virtual ICollection<ProductImageModel>? ProductImages { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: This Collection can be empty, not null

File: Curtains/Curtains.Domain/Models/BedspreadsModel.cs [hunk 23]
```diff
@@ -31,40 +19,24 @@ public BedspreadsModel()
...
         /// <summary>
-        ///  This property represents list of UserOrders, in which there are bedspreads.
+        ///  This property represents product sets with bedspreads.
         /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
+        public virtual ICollection<ProductSetModel>? ProductSets { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/BedspreadsModel.cs]: The same

File: Curtains/Curtains.Domain/Models/BracingModel.cs [hunk 24]
```diff
@@ -0,0 +1,14 @@
...
+{
+    /// <summary>
+    ///  This class describes accessories model.
+    /// </summary>
+    public class BracingModel : ExtendedBaseEntity
+    {
+        #region NavigationProperties
+        public virtual ICollection<CurtainModel>? Curtains { get; }
+
+        public virtual ICollection<LambrequinsModel>? Lambrequins { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/BracingModel.cs]: The same

File: Curtains/Curtains.Domain/Models/ColorModel.cs [hunk 25]
```diff
@@ -1,43 +1,31 @@
...
         ///  This property represents list of accessories with that color.
         /// </summary>
-        public virtual ICollection<AccessoriesModel> Accessories { get; set; }
+        public virtual ICollection<AccessoriesModel>? Accessories { get; }
         /// <summary>
-        ///  This property represents list of lamrequins with that color.
+        ///  This property represents list of Curtains with that color.
         /// </summary>
-        public virtual ICollection<LambrequinsModel> Lambrequins { get; set; }
+        public virtual ICollection<CurtainModel>? Curtains { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ColorModel.cs]: The same

File: Curtains/Curtains.Domain/Models/ConsistencyModel.cs [hunk 26]
```diff
@@ -1,26 +1,12 @@
...
-        {
-            Fabrics = new HashSet<FabricModel>();
-        }
-
         #region NavigationPropertiesRegion
         /// <summary>
         ///  This property represents list of fabrics with that consistency.
         /// </summary>
-        public virtual ICollection<FabricModel> Fabrics { get; set; }
+        public virtual ICollection<FabricModel>? Fabrics { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ConsistencyModel.cs]: The same

File: Curtains/Curtains.Domain/Models/CurtainsKindModel.cs [hunk 27]
```diff
@@ -2,19 +2,11 @@
...
-        {
-            Curtains = new HashSet<CurtainsModel>();
-        }
-
         #region NavigationPropertiesRegion
         /// <summary>
         ///  This property represents list of curtains with that kind.
         /// </summary>
-        public virtual ICollection<CurtainsModel> Curtains { get; set; }
+        public virtual ICollection<CurtainModel>? Curtains { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/CurtainsKindModel.cs]: Why did you removed constructor and set?

File: Curtains/Curtains.Domain/Models/CurtainsModel.cs [hunk 28]
```diff
@@ -86,17 +75,25 @@ public CurtainsModel()
...
+        ///  This property represents Lambrequins with that curtains.
         /// </summary>
-        public virtual ICollection<ProductSetModel> Sets { get; set; }
+        public virtual ICollection<LambrequinsModel>? Lambrequins { get; }
         /// <summary>
-        ///  This property represents list of UserOrders with that curtains.
+        ///  This property represents product images with that curtains.
         /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
+        public virtual ICollection<ProductImageModel>? ProductImages { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/CurtainsModel.cs]: Same

File: Curtains/Curtains.Domain/Models/CurtainsTypeModel.cs [hunk 29]
```diff
@@ -1,26 +1,15 @@
...
-        {
-            Curtains = new HashSet<CurtainsModel>();
-        }
-
         #region NavigationPropertiesRegion
         /// <summary>
         ///  This property represents list of curtains of that type.
         /// </summary>
-        public virtual ICollection<CurtainsModel> Curtains { get; set; }
+        public virtual ICollection<CurtainModel>? Curtains { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/CurtainsTypeModel.cs]: same

File: Curtains/Curtains.Domain/Models/DecorationsModel.cs [hunk 30]
```diff
@@ -27,7 +16,7 @@ public DecorationsModel()
         /// <summary>
         ///  This property represents list of fabrics with that decorations.
         /// </summary>
-        public virtual ICollection<FabricModel> Fabrics { get; set; }
+        public virtual ICollection<FabricModel>? Fabrics { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/DecorationsModel.cs]: same

File: Curtains/Curtains.Domain/Models/DesignModel.cs [hunk 31]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Curtains.Domain.Models
+{
+    public class DesignModel : ExtendedByTitleBaseEntity
+    {
+        #region NavigationPropertiesRegion
+        /// <summary>
+        ///  This property represents list of fabrics of that design.
+        /// </summary>
+        public virtual ICollection<FabricModel>? Fabrics { get; }
+        #endregion
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/DesignModel.cs]: same

File: Curtains/Curtains.Domain/Models/FabricKindModel.cs [hunk 32]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        /// <summary>
+        ///  This property represents list of fabrics with that kind.
+        /// </summary>
+        public virtual ICollection<FabricModel>? Fabrics { get; }
+        /// <summary>
+        ///  This property represents list of lambrequins with that kind.
+        /// </summary>
+        public virtual ICollection<LambrequinsModel>? Lambrequins { get; }
+    }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/FabricKindModel.cs]: Same

File: Curtains/Curtains.Domain/Models/FabricModel.cs [hunk 33]
```diff
@@ -66,23 +65,46 @@ public FabricModel()
...
+        public virtual ICollection<BedspreadsModel>? Bedspreads { get; set; }
         /// <summary>
         ///  This property represents list of pillows with that fabric.
         /// </summary>
-        public virtual ICollection<PillowsModel> Pillows { get; set; }
+        public virtual ICollection<PillowsModel>? Pillows { get; set; }
+        /// <summary>
+        ///  This property represents product images of fabric.
+        /// </summary>
+        public virtual ICollection<ProductImageModel>? ProductImages { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/FabricModel.cs]: same

File: Curtains/Curtains.Domain/Models/FabricTypeModel.cs [hunk 34]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Curtains.Domain.Models
+{
+    public class FabricTypeModel : ExtendedByTitleBaseEntity
+    {
+        /// <summary>
+        ///  This property represents list of fabrics with that type.
+        /// </summary>
+        public virtual ICollection<FabricModel>? Fabrics { get; }
+    }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/FabricTypeModel.cs]: same

File: Curtains/Curtains.Domain/Models/LambrequinsModel.cs [hunk 35]
```diff
@@ -33,9 +38,17 @@ public LambrequinsModel()
...
+        public virtual BracingModel Bracing { get; set; }
+        /// <summary>
+        ///  This property represents fabricKind of lambrequins.
+        /// </summary>
+        public virtual FabricKindModel FabricKind { get; set; }
+        /// <summary>
+        ///  This property represents product sets with this lambrequin.
         /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
+        public virtual ICollection<ProductSetModel>? ProductSets { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/LambrequinsModel.cs]: same

File: Curtains/Curtains.Domain/Models/ManufacturerModel.cs [hunk 36]
```diff
@@ -27,7 +16,7 @@ public ManufacturerModel()
         /// <summary>
         ///  This property represents list of fabrics with that manufacturer.
         /// </summary>
-        public virtual ICollection<FabricModel> Fabrics { get; set; }
+        public virtual ICollection<FabricModel>? Fabrics { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ManufacturerModel.cs]: same

File: Curtains/Curtains.Domain/Models/MarketingInfoModel.cs [hunk 37]
```diff
@@ -25,5 +25,12 @@ public class MarketingInfoModel : BaseEntity
         /// </summary>
         public bool IsPopular { get; set; }
         #endregion
+
+        #region NavigationProperties
+        /// <summary>
+        ///  This property represents ProductImage.
+        /// </summary>
+        public virtual ProductImageModel ProductImage { get; set; }
+        #endregion
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/MarketingInfoModel.cs:45]: Why did you add this property?

File: Curtains/Curtains.Domain/Models/MaterialModel.cs [hunk 38]
```diff
@@ -1,26 +1,19 @@
...
         #region NavigationPropertiesRegion
         /// <summary>
         ///  This property represents list of curtains with that material.
         /// </summary>
-        public virtual ICollection<CurtainsModel> Curtains { get; set; }
+        public virtual ICollection<CurtainModel>? Curtains { get; }
+        /// <summary>
+        ///  This property represents list of fabrics with that material.
+        /// </summary>
+        public virtual ICollection<FabricModel>? Fabrics { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/MaterialModel.cs]: same

File: Curtains/Curtains.Domain/Models/PillowsModel.cs [hunk 39]
```diff
@@ -50,17 +34,13 @@ public PillowsModel()
...
+        public virtual ICollection<ProductImageModel>? ProductImages { get; }
         /// <summary>
         ///  This property represents list of product sets with that pillows.
         /// </summary>
-        public virtual ICollection<ProductSetModel> Sets { get; set; }
-        /// <summary>
-        ///  This property represents list of UserOrders with that pillows.
-        /// </summary>
-        public virtual ICollection<UserOrderModel> UserOrders { get; set; }
+        public virtual ICollection<ProductSetModel>? ProductSets { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/PillowsModel.cs]: same

File: Curtains/Curtains.Domain/Models/ProductImageModel.cs [hunk 40]
```diff
@@ -16,29 +10,54 @@ public class ProductImageModel : BaseEntity
...
+        public int? BedspreadsId { get; set; }
+        /// <summary>
+        ///  This property represents SetsId.
+        /// </summary>
+        public int? SetsId { get; set; }
+        /// <summary>
+        ///  This property represents FabricId.
+        /// </summary>
+        public int? FabricId { get; set; }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ProductImageModel.cs:33]: Why nullable?

File: Curtains/Curtains.Domain/Models/ProductImageModel.cs [hunk 41]
```diff
@@ -16,29 +10,54 @@ public class ProductImageModel : BaseEntity
...
         /// <summary>
         ///  This property represents product sets, which image is presented.
         /// </summary>
-        public virtual ProductSetModel Sets { get; set; }
+        public virtual ProductSetModel? Sets { get; set; }
         /// <summary>
         ///  This property represents fabrics, which image is presented.
         /// </summary>
-        public virtual FabricModel Fabrics { get; set; }
+        public virtual FabricModel? Fabrics { get; set; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ProductImageModel.cs]: Why nullable?

File: Curtains/Curtains.Domain/Models/ProductSetModel.cs [hunk 42]
```diff
@@ -34,15 +32,23 @@ public class ProductSetModel : ExtendedBaseEntity
...
-        public virtual BedspreadsModel Bedspreads { get; set; }
+        public virtual BedspreadsModel? Bedspreads { get; set; }
+        /// <summary>
+        ///  This property represents lambrequin in that set.
+        /// </summary>
+        public virtual LambrequinsModel? Lambrequins { get; set; }
+        /// <summary>
+        ///  This property represents product images with this sets.
+        /// </summary>
+        public virtual ICollection<ProductImageModel>? ProductImages { get; }
```
Comment (zd-prog) [inline Curtains/Curtains.Domain/Models/ProductSetModel.cs]: Check all models for nullable collections

File: Curtains/Curtains.Infrastructure/Database/CurtainsDbContext.cs [hunk 43]
```diff
@@ -12,26 +12,26 @@ public class CurtainsDbContext : DbContext
         private IDbContextTransaction _currentTransaction;
 
         #region DbSetsRegion
-        public virtual DbSet<AccessoriesModel> Accessories { get; set; } 
-        public virtual DbSet<ColorModel> Colors { get; set; } 
-        public virtual DbSet<CurtainsModel> Curtains { get; set; } 
+        public virtual DbSet<AccessoriesModel> Accessories { get; set; }
+        public virtual DbSet<ColorModel> Colors { get; set; }
+        public virtual DbSet<CurtainModel> Curtains { get; set; }
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Database/CurtainsDbContext.cs:17]: Why did you rename model?

File: Curtains/Curtains.Infrastructure/Repositories/AccessoriesRepository.cs [hunk 44]
```diff
@@ -39,6 +32,12 @@ public AccessoriesRepository(CurtainsDbContext curtainsContext, ILogger logger)
         /// <returns>Collection of AccessoriesModel entities in List type</return>
         public IEnumerable<AccessoriesModel> GetAll()
         {
+            if (!_curtainsContext.Accessories.Any())
+            {
+                _logger.LogError("Accessories table is empty");
+                throw new ResourceNotFoundException();
+            }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/AccessoriesRepository.cs:40]: For what reason you throw exception, if we can return empty list

File: Curtains/Curtains.Infrastructure/Repositories/BedspreadsRepository.cs [hunk 45]
```diff
@@ -36,6 +33,12 @@ public BedspreadsRepository(CurtainsDbContext curtainsContext, ILogger logger)
         /// <returns>Collection of BedspreadsModel entities in List type</return>
         public IEnumerable<BedspreadsModel> GetAll()
         {
+            if (!_curtainsContext.Bedspreads.Any())
+            {
+                _logger.LogError("Bedspreads table is empty");
+                throw new ResourceNotFoundException();
+            }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/BedspreadsRepository.cs:41]: same

File: Curtains/Curtains.Infrastructure/Repositories/BracingRepository.cs [hunk 46]
```diff
@@ -0,0 +1,127 @@
...
+        /// This method get all<c> BracingModel <c> entities from database with includes tabels by entity Id
+        /// </summary>
+        /// <returns>Collection of BracingModel entities in List type</return>
+        public IEnumerable<BracingModel> GetAll()
+        {
+            if (!_curtainsContext.Bracings.Any())
+            {
+                _logger.LogError("Bracings table is empty");
+                throw new ResourceNotFoundException();
+            }
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/BracingRepository.cs:39]: same

File: Curtains/Curtains.Infrastructure/Repositories/ColorRepository.cs [hunk 47]
```diff
@@ -36,6 +37,12 @@ public ColorRepository(CurtainsDbContext curtainsContext, ILogger logger)
         /// <returns>Collection of ColorModel entities in List type</return>
         public IEnumerable<ColorModel> GetAll()
         {
+            if (!_curtainsContext.Colors.Any())
+            {
+                _logger.LogError("Colors table is empty");
+                throw new ResourceNotFoundException();
+            }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/ColorRepository.cs:45]: same

File: Curtains/Curtains.Infrastructure/Repositories/ConsistencyRepository.cs [hunk 48]
```diff
@@ -36,6 +32,12 @@ public ConsistencyRepository(CurtainsDbContext curtainsContext, ILogger logger)
         /// <returns>Collection of ConsistencyModel entities in List type</return>
         public IEnumerable<ConsistencyModel> GetAll()
         {
+            if (!_curtainsContext.Consistencies.Any())
+            {
+                _logger.LogError("Consistencies table is empty");
+                throw new ResourceNotFoundException();
+            }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/ConsistencyRepository.cs:40]: same

File: Curtains/Curtains.Infrastructure/Repositories/CurtainsRepository.cs [hunk 49]
```diff
@@ -34,8 +30,14 @@ public CurtainsRepository(CurtainsDbContext curtainsContext, ILogger logger)
...
         /// <returns>Collection of CurtainsModel entities in List type</return>
-        public IEnumerable<CurtainsModel> GetAll()
+        public IEnumerable<CurtainModel> GetAll()
         {
+            if (!_curtainsContext.Curtains.Any())
+            {
+                _logger.LogError("Curtains table is empty");
+                throw new ResourceNotFoundException();
+            }
+
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Repositories/CurtainsRepository.cs:40]: Check all repositories

**Review by zd-prog — CHANGES_REQUESTED** [review]

File: Curtains/Curtains.Domain/Models/ProductImageModel.cs (hunk 40 above)
Comment (C0Rs1K) [reply Curtains/Curtains.Domain/Models/ProductImageModel.cs:33]: cuz it can be null in db

File: Curtains/Curtains.Infrastructure/Repositories/AccessoriesRepository.cs (hunk 44 above)
Comment (C0Rs1K) [reply Curtains/Curtains.Infrastructure/Repositories/AccessoriesRepository.cs:40]: in other methods we throw exception

File: Curtains/Curtains.Domain/Models/MarketingInfoModel.cs (hunk 37 above)
Comment (C0Rs1K) [reply Curtains/Curtains.Domain/Models/MarketingInfoModel.cs:45]: it shows the product image it is connected with

**Comment by EgorAntonovich** [comment]
@zd-prog waiting for your approve

**Review by zd-prog — APPROVED** [review]

**Comment by EgorAntonovich** [comment]
@C0Rs1K waiting for your merge
